src/App.tsx

```diff
@@ -14,11 +15,127 @@ const goodsFromServer: string[] = [
   'Garlic',
 ];
 
-const App: React.FC = () => (
-  <div className="App">
-    <h1>Goods</h1>
-    {goodsFromServer.length}
-  </div>
-);
+type State = {
+  goods: string[],
+  isVisible: boolean,
+  isReversed: boolean,
+  sortBy: string,
+};
+
+class App extends React.Component<{}, State> {
+  state = {
+    goods: goodsFromServer,
+    isVisible: false,
+    isReversed: false,
+    sortBy: '',
```

Comment:
you don't have to save these values in state, make reverse and when on button click